source/postprocess/visualization.cc

```diff
@@ -320,6 +329,16 @@ namespace aspect
           && (this->get_timestep_number() != 0))
         return std::pair<std::string,std::string>();
 
+      // up the counter of the number of the file by one, but not in
+      // the very first output step. if we run postprocessors on all
+      // iterations, only increase file number in the first nonlinear iteration
+      const bool first_output = (this->get_timestep_number() == 0)
```

Comment:
That makes a lot of sense, thanks!
I wasn't happy with my original version, but didn't have a good fix.